deploy.py

```diff
@@ -13,105 +13,86 @@ def add_cli_args():
-                             'e.g. zuul01, nodepool02, allinone01.',
-                        )
-    parser.add_argument('--with-vars',
-                        action='store_true',
-                        help='show the hosts graph of ansible inventory with vars, specify with '
-                             'key-value pairs combined with "="',
-                        )
-    parser.add_argument('--graph',
-                        action='store_true',
-                        help='show the hosts graph of ansible inventory',
-                        )
-    parser.add_argument('--show-old-ips',
-                        action='store_true',
-                        help='show the old hosts ip address',
-                        )
-    parser.add_argument('--show-new-ips',
-                        action='store_true',
-                        help='show the new hosts ip address',
+                        help='specify new ip address for current inventory, '
+                             'this argument need to use with key-value pairs '
+                             'combined with "=", the key must be the host name'
+                             ' defined in the inventory yaml files, e.g. '
+                             'zuul01, nodepool02, allinone01.',
                         )
     return parser
 
 
 def main():
     parsed_args = add_cli_args().parse_args()
     os.environ['OL_TYPE'] = parsed_args.type
-    cmd = ['ansible-playbook', '-i', 'inventory/inventory.py', 'playbooks/site.yaml']
-
-    if parsed_args.user:
-        cmd += ['-u', parsed_args.user]
+    cmd = []
+
+    if parsed_args.action == 'deploy':
+        cmd = ['ansible-playbook', '-i', 'inventory/inventory.py',
+               'playbooks/site.yaml']
+    elif parsed_args.action == 'new-slave':
+        cmd = ['ansible-playbook', '-i', 'inventory/inventory.py',
+               'playbooks/site.yaml', '-l', '*-slave']
+        parsed_args.extra_vars.append('config_new_slave=true')
+    elif parsed_args.action == 'switch-role':
+        os.environ['OL_SWITCH_MASTER_SLAVE'] = True
+        cmd = ['ansible-playbook', '-i', 'inventory/inventory.py',
+               'playbooks/conf-new-slave.yaml']
+    elif parsed_args.action == 'show-graph':
+        cmd = ['ansible-inventory', '-i', 'inventory/inventory.py', '--graph']
+    elif parsed_args.action == 'show-ip':
+        cmd = ['python', 'inventory/inventory.py', '--show-ip']
 
-    if parsed_args.new_ip:
-        specified_ips = dict([(d.partition('=')[::2]) for d in parsed_args.new_ip])
+    if parsed_args.host_ip:
+        specified_ips = dict([(d.partition('=')[::2])
+                              for d in parsed_args.new_ip])
         for host, ip in specified_ips.items():
             os.environ['OL_%s_IP' % host.upper()] = ip
 
-    if parsed_args.new_slave:
-        if not parsed_args.new_ip:
-            sys.exit("ERROR: Must specify the new slave hosts IPs with --new-ip")
```

Comment:
please add this limit

Reply:
I prefer not, IMO If new_ip is provided, use it. If not, just use the IPs in inventory file. All my change for this PR is to make the cli as clean as it could

Reply:
it is easy to forget to update the IPs in inventory files, in new slave situation. what we added this cli tool is to get rid of manually modification of the inventory files. so I think this is required.
I do not thinks the shorter help messages can make things better... and cleaner

Reply:
1. Update the inventory file should be the first step when first time using `labkeeper` IMO.
2. `new_slave` action is not always happen,(maybe 1-6 months or even 1 year). Once using it, I assume users will always download the newest `labkeeper`, Then I can't imaging that `update the inventory file` is easy to be forgotten. Users will use a file or a tool that is updated long time ago? Amazing.
3. Keep the same behavior with `deploy` action, when deploy, the `new_ip` is not required.

Reply:
1. that is the old way, and that is why I add this deploy.py, to provide a way to avoid modify the inventory files, so it is not necessary now.
2. so you only need to update the repo. do not need to modify any things, and once a mistake happened running the `new_slave` without update the inventory, it will re-deploy the slave, that may cause bad effect. 
3. why ?

Reply:
I don't mean to remove `new_ip`, I mean we should keep `new_slave` the same behavior with `deploy`. If it's required, it should be both required for these actions. You said we should not rely on  inventory, why `deploy` action doesn't force require `new_ip`? Either make it required or make it optional is OK, but it should be the same for all related actions.

Reply:
I'm afraid I cant stop myself to click "merge" for coming deadline. ;-)

Reply:
@wangxiyuan actually, at first, I also thought like you and did't made that argument required, but after some testing operations,  I changed the `--new-ip` required because I think it should be necessary, if you run new-slave and forget updating IPs. that may be a serious, because it will re-install the slave, maybe it will cause some data missing. if you make the `--new-ip` required. it will compulsively operators and let him know it will deploy the new slave part in two new nodes specified by the `--new-ip`.